src/main/java/org/jenkins/plugins/lockableresources/LockStep.java

```diff
@@ -11,25 +11,45 @@
 import hudson.Extension;
 import hudson.model.AutoCompletionCandidates;
 
+import edu.umd.cs.findbugs.annotations.Nullable;
+
 public class LockStep extends AbstractStepImpl implements Serializable {
 
-	public final String resource;
+	public String resource = null;
+
+	public String label = null;
+
+	public int quantity = 0;
 
 	public boolean inversePrecedence = false;
 
 	@DataBoundConstructor
-	public LockStep(String resource) {
-		if (resource == null || resource.isEmpty()) {
-			throw new IllegalArgumentException("must specify resource");
-		}
-		this.resource = resource;
+	public LockStep() {
 	}
 
 	@DataBoundSetter
 	public void setInversePrecedence(boolean inversePrecedence) {
 		this.inversePrecedence = inversePrecedence;
 	}
 
+	@DataBoundSetter
+	public void setResource(@Nullable String resource) {
+		this.resource = resource;
+	}
+
+	@DataBoundSetter
+	public void setLabel(@Nullable String label) {
+		if (label != null && !label.isEmpty())
+		{
```

Comment:
Use inlined `{` please:

```
if (...) {
    ...
} else {
    ...
}
```
I'm preparing contributing guidelines to define this.